core/src/main/java/com/aem/demo/core/models/MainPageModel.java

```diff
@@ -100,4 +106,27 @@ private List<NewsCardModel> separateModels(List<NewsCardModel> list, int paginat
         return separatedPart;
     }
 
+    private void paginateModels(List<NewsCardModel> models) {
+        double listSize = models.size();
+        double listPart = listSize / 5;
+        double roundPart = Math.ceil(listPart);
+
+        pageNumbers = IntStream.range(1, (int) (roundPart + 1)).boxed().collect(Collectors.toList());
+
+    }
+
+    private void isNumeric(String string) {
+        int intValue;
```

Comment:
this value is never used. pls remove redundant assignment

Reply:
See @Zubayr99 you've created a perfect sample of a method that does.... nothing!)) 

It counts the `intValue`. But does not return it to no one.
It catches an exception. But does not prevent all the same exception to happen in line 85.
And, whether this method is called, or not called, the computation of number is re-done later on in line 85 and 89.

Please refactor this method